### PR TITLE
Add DrawingArea for FamilyInstance

### LINKED ISSUE #1388: Add `DrawingArea` working with family instances
#### Description:

As in the title. Needed for the existing titleblock family instances in the model.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1388 

Added method returning drawing area outline from the title block family instance.

### CODE REVIEW

File: Revit_Core_Engine/Query/DrawingArea.cs [hunk 1]
```diff
@@ -37,7 +37,32 @@ public static partial class Query
...
 
-        [Description("Compute the outline of Title Block drawing area.")]
+        [Description("Returns the drawing area outline of the Title Block instance.")]
+        [Input("titleBlock", "Title Block to get the drawing area outline from.")]
+        [Output("outline", "The Title Block's drawing area.")]
+        public static Outline DrawingArea(this FamilyInstance titleBlock)
+        {
+            if (titleBlock == null || !((BuiltInCategory)titleBlock.Category.Id.IntegerValue == Autodesk.Revit.DB.BuiltInCategory.OST_TitleBlocks))
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Title block cannot be null and has to be of Title Block category.");
```
Comment (vietle-bh) [inline Revit_Core_Engine/Query/DrawingArea.cs:47]:
```suggestion
                BH.Engine.Base.Compute.RecordWarning($"Title block cannot be null and has to be of the Title Block category.");
```

File: Revit_Core_Engine/Query/DrawingArea.cs [hunk 2]
```diff
@@ -37,7 +37,32 @@ public static partial class Query
...
+                return null;
+            }
+
+            Outline drawingArea = titleBlock.Symbol.DrawingArea();
+            Transform transform = titleBlock.GetTotalTransform();
+
+            if (!transform.IsIdentity)
+            {
+                drawingArea.MaximumPoint += transform.Origin;
+                drawingArea.MinimumPoint += transform.Origin;
```
Comment (vietle-bh) [inline Revit_Core_Engine/Query/DrawingArea.cs:57]: I think we should use all data in the Transform, even if it's only a movement like in this case.
```suggestion
                drawingArea.MaximumPoint = transform.OfPoint(drawingArea.MaximumPoint);
                drawingArea.MinimumPoint = transform.OfPoint(drawingArea.MinimumPoint);
```

**Review by vietle-bh — DISMISSED** [review]
No more comments. Let's keep going 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check required

File: Revit_Core_Engine/Query/DrawingArea.cs [hunk 3]
```diff
@@ -37,7 +37,32 @@ public static partial class Query
...
 
-        [Description("Compute the outline of Title Block drawing area.")]
+        [Description("Returns the drawing area outline of the Title Block instance.")]
+        [Input("titleBlock", "Title Block to get the drawing area outline from.")]
+        [Output("outline", "The Title Block's drawing area.")]
+        public static Outline DrawingArea(this FamilyInstance titleBlock)
+        {
+            if (titleBlock == null || !((BuiltInCategory)titleBlock.Category.Id.IntegerValue == Autodesk.Revit.DB.BuiltInCategory.OST_TitleBlocks))
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Title block cannot be null and has to be of the Title Block category.");
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/DrawingArea.cs:47]: Why not `RecordError`?

Comment (michal-pekacki) [reply Revit_Core_Engine/Query/DrawingArea.cs:47]: Good catch! It should be `RecordError`.

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Review by pawelbaran — APPROVED** [review]
Happy with the changes now 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: versioning, ready-to-merge

**Pull request merged**